mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/search/DeletedMessageField.java

```diff
@@ -0,0 +1,53 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.search;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.vault.DeletedMessage;
+
+public interface DeletedMessageField<T> {
+    interface ValueExtractor<T> {
+        Optional<T> extract(DeletedMessage deletedMessage);
+
+        ValueExtractor<ZonedDateTime> DELETION_DATE_EXTRACTOR = deletedMessage -> Optional.ofNullable(deletedMessage.getDeletionDate());
```

Comment:
why should we handle null case ? Does a deleted message without a deletion date exist at all?
Same question for other fields.

Reply:
DeletionDate is an compulsory field, base on the DeletedMessage model, Subject or Sender could be missing. then I propose an Optional returned for field extractors. Otherwise I cannot set a default value to subject or sender because they are actually missing

Reply:
ok but don't use `ofNullable`, use `of` instead